src/Graphics/Vty/Input/Loop.hs

```diff
@@ -128,19 +131,20 @@ readFromDevice = do
         appliedConfig .= newConfig
     bufferPtr <- use $ inputBuffer.ptr
     maxBytes  <- use $ inputBuffer.size
-    stringRep <- liftIO $ do
-        -- The killThread used in shutdownInput will not interrupt the
-        -- foreign call fdReadBuf uses this provides a location to be
-        -- interrupted prior to the foreign call. If there is input on
-        -- the FD then the fdReadBuf will return in a finite amount of
-        -- time due to the vtime terminal setting.
-        threadWaitRead fd
-        bytesRead <- fdReadBuf fd bufferPtr (fromIntegral maxBytes)
-        if bytesRead > 0
-        then map (chr . fromIntegral) <$> peekArray (fromIntegral bytesRead) bufferPtr
-        else return []
-    when (not $ null stringRep) $ logMsg $ "input bytes: " ++ show stringRep
-    return stringRep
+    -- The killThread used in shutdownInput will not interrupt the
+    -- foreign call fdReadBuf uses this provides a location to be
+    -- interrupted prior to the foreign call. If there is input on
+    -- the FD then the fdReadBuf will return in a finite amount of
+    -- time due to the vtime terminal setting.
+    liftIO $ threadWaitRead fd
```

Comment:
This restructuring is not great in hindsight, I'll put this back as a `liftIO $ do` block.

Reply:
Done.